src/server.cpp

```diff
@@ -306,6 +306,11 @@ Server::~Server()
 	for (auto &detached_inventory : m_detached_inventories) {
 		delete detached_inventory.second;
 	}
+
+	while (!m_unsent_map_edit_queue.empty()) {
```

Comment:
This fixes the memory leak, but the data lost here is negligible. The rest of the map changes are to improve allocation safety.